misk-hibernate-testing/src/main/kotlin/misk/jdbc/AbstractScaleSafetyChecks.kt

```diff
@@ -0,0 +1,154 @@
+package misk.jdbc
+
+import com.squareup.moshi.Moshi
+import misk.hibernate.Transacter
+import misk.moshi.adapter
+import misk.okio.split
+import misk.vitess.StartDatabaseService
+import mu.KotlinLogging
+import java.sql.Connection
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okio.BufferedSource
+import okio.ByteString.Companion.encodeUtf8
+import java.io.EOFException
+import java.sql.Timestamp
+import javax.inject.Inject
+
+internal data class Instruction(val Opcode: String, val Input: Instruction?) {
+  val isScatter: Boolean
+    get() {
+      return Opcode == "SelectScatter" || Input?.isScatter ?: false
+    }
+}
+
+internal data class QueryPlan(
+  val Original: String,
+  val Instructions: Instruction,
+  val ExecCount: Int
+) {
+  val isScatter: Boolean
+    get() = Instructions.isScatter
+}
+
+internal data class Variables(
+  /** Queries processed by plan type */
+  val QueriesProcessed: Map<String, Int>
+)
+
+abstract class AbstractScaleSafetyChecks @Inject constructor(
```

Comment:
Rather than reuse by inheritance can we reuse by composition? Inheritance is a blunt tool for reuse and one that I prefer to avoid...

Reply:
Actually I think these could even be static methods where you pass in the connection as a parameter